src/plugins/dbusrecv/testmod_dbusrecv.c

```diff
@@ -0,0 +1,253 @@
+
+	char * expectedKeyName = "system/tests/testmod_dbusrecv/added";
+	dbusSendMessage ("KeyAdded", expectedKeyName);
+
+	ElektraIoTimerOperation * timerOp = elektraIoNewTimerOperation (TEST_TIMEOUT, 1, test_timerCallback, NULL);
+	elektraIoBindingAddTimer (binding, timerOp);
+
+	test_callbackKey = NULL;
+	test_callbackLoop = loop;
+	uv_run (loop, UV_RUN_DEFAULT);
+
+	succeed_if_same_string (expectedKeyName, keyName (test_callbackKey));
+
+	// close notification
+	func = elektraPluginGetFunction (plugin, "closeNotification");
+	exit_if_fail (func, "could not get function closeNotification");
+	ElektraNotificationCloseNotification closeNotification = (ElektraNotificationCloseNotification)func;
+	closeNotification (plugin);
+
+	elektraIoBindingRemoveTimer (timerOp);
+	elektraFree (timerOp);
+	keyDel (test_callbackKey);
+	PLUGIN_CLOSE ();
+}
+
+static void test_keyChanged (uv_loop_t * loop, ElektraIoInterface * binding)
+{
+	printf ("test adding keys\n");
+
+	KeySet * conf = ksNew (0, KS_END);
+	PLUGIN_OPEN ("dbusrecv");
+
+	// io binding is required for dispatching
+	size_t func = elektraPluginGetFunction (plugin, "setIoBinding");
+	exit_if_fail (func, "could not get function setIoBinding");
+	ElektraIoPluginSetBinding setIoBinding = (ElektraIoPluginSetBinding)func;
+	setIoBinding (plugin, binding);
+
+	// open notification
+	func = elektraPluginGetFunction (plugin, "openNotification");
+	exit_if_fail (func, "could not get function openNotification");
+	ElektraNotificationOpenNotification openNotification = (ElektraNotificationOpenNotification)func;
+	openNotification (plugin, test_notificationCallback, NULL);
+
+	char * expectedKeyName = "system/tests/testmod_dbusrecv/changed";
+	dbusSendMessage ("KeyChanged", expectedKeyName);
+
+	ElektraIoTimerOperation * timerOp = elektraIoNewTimerOperation (TEST_TIMEOUT, 1, test_timerCallback, NULL);
+	elektraIoBindingAddTimer (binding, timerOp);
+
+	test_callbackKey = NULL;
+	test_callbackLoop = loop;
+	uv_run (loop, UV_RUN_DEFAULT);
+
+	succeed_if_same_string (expectedKeyName, keyName (test_callbackKey));
+
+	// close notification
+	func = elektraPluginGetFunction (plugin, "closeNotification");
+	exit_if_fail (func, "could not get function closeNotification");
+	ElektraNotificationCloseNotification closeNotification = (ElektraNotificationCloseNotification)func;
+	closeNotification (plugin);
+
+	elektraIoBindingRemoveTimer (timerOp);
+	elektraFree (timerOp);
+	keyDel (test_callbackKey);
+	PLUGIN_CLOSE ();
+}
+
+int main (int argc, char ** argv)
+{
+	printf ("DBUSRECV TESTS\n");
+	printf ("==============\n\n");
+
+	init (argc, argv);
+
+	// Test if dbus is available
+	test_prerequisites ();
+
+	uv_loop_t * loop = uv_default_loop ();
+	ElektraIoInterface * binding = elektraIoUvNew (loop);
+
+	// Test added, changed & deleted
+	test_keyAdded (loop, binding);
+	test_keyChanged (loop, binding);
+
+	printf ("\ntestmod_dbusrecv RESULTS: %d test(s) done. %d error(s).\n", nbTest, nbError);
```

Comment:
@sanssecours wrote a print_result which makes the output the same across all tests.